src/rshim.c

```diff
@@ -524,13 +525,18 @@ static void rshim_mmio_write_common(rshim_backend_t *bd, uintptr_t pa,
         (1ULL << RSH_MEM_ACC_CTL__WRITE_SHIFT) |
         (1ULL << RSH_MEM_ACC_CTL__SEND_SHIFT);
   bd->write_rshim(bd, RSHIM_CHANNEL, RSH_MEM_ACC_CTL, reg);
-  rshim_reg_indirect_wait(bd, resp_count);
+  return rshim_reg_indirect_wait(bd, resp_count);
 }
 
-uint64_t rshim_mmio_read_common(rshim_backend_t *bd, uintptr_t pa, uint8_t size)
+uint64_t rshim_mmio_read_common(rshim_backend_t *bd, uintptr_t pa,
+                                uint8_t size, uint64_t *data)
 {
   uint64_t reg, resp_count;
 
+  bd->read_rshim(bd, RSHIM_CHANNEL, RSH_DEVICE_MSTR_PRIV_LVL, &reg);
+  reg |= 0x1ULL << RSH_DEVICE_MSTR_PRIV_LVL__MEM_ACC_LVL_SHIFT;
+  bd->write_rshim(bd, RSHIM_CHANNEL, RSH_DEVICE_MSTR_PRIV_LVL, reg);
```

Comment:
Add a check to only do this for BF2 (since BF1 is different)

Reply:
Not sure I understand this one.  The mem acc widget security level setting is the same between BF1/BF2.  The name of the register changed, but the location and meaning are the same.

Reply:
Please ignore this comment. Yes, the offset and bit are the same for both BF1 anf BF2.